src/main/kotlin/sipura/alg/Connectivity.kt

```diff
@@ -163,11 +170,119 @@ object Connectivity {
     }
 
     /**
+     * Calculates the distance between the two given vertices [v1] and [v2].
+     *
+     * The distance is defined as the number of edges in the shortest path between the two vertices.
      * The distance of two neighboring vertices is 1, and the distance of a vertex to itself is 0.
      *
+     * Runtime: O(n + m)
      * @throws IllegalArgumentException if [v1] or [v2] is not in the graph.
-     *
-     * @return The number of edges in the shortest path from [v1] to [v2] in graph [g]
+     * @return The distance between [v1] and [v2].
      */
     fun <V> distance(g: SimpleGraph<V>, v1: V, v2: V): Int = shortestPath(g, v1, v2).size - 1
+
+    /**
+     * Calculates all cut-vertices and all bridge-edges of the given graph [g].
+     *
+     * A vertex v is a cut-vertex if the graph has more connected components after v is removed from it.
+     * An edge (v1,v2) is a bridge-edge if the graph has more connected components after (v1,v2) is removed from it.
+     *
+     * Runtime: O(n + m)
+     * @return A [Pair] containing a set of all cut-vertices and a set of all bridge-edges.
+     */
+    fun <V> cutVerticesAndBridgeEdges(g: SimpleGraph<V>): Pair<Set<V>, Set<Pair<V, V>>> {
+        val cutVertices = HashSet<V>()
+        val bridgeEdges = HashSet<Pair<V, V>>()
+        val remaining = g.V.toHashSet()
+        // make sure we visit all connected components
+        while (remaining.isNotEmpty()) {
```

Comment:
can you add to the docs with algorithm is used?

Reply:
also, are you sure we can't reuse some other parts of the library here?

Reply:
what do you think I could reuse?